.github/actions/pre_cache/action.yml

```diff
@@ -0,0 +1,43 @@
+name: "Pre-Cache"
+runs:
+  using: composite
+  steps:
+    - name: Get 'make' checksum
+      run: |
+        cp -f /usr/bin/make make.bin
+        md5sum make.bin > $GITHUB_WORKSPACE/make.md5
+      shell: bash -ileo pipefail {0}
+    - name: Get 'dot' checksum
+      run: |
+        if [ -f /usr/bin/dot ] ; then
+          cp -f /usr/bin/dot dot.bin
+          md5sum dot.bin > $GITHUB_WORKSPACE/dot.md5
+        else
+          touch dot.bin
+          md5sum dot.bin > $GITHUB_WORKSPACE/dot.md5
+          rm -f dot.bin
+        fi
```

Comment:
Same comment as down below in the cache key construction for the python env.

Can't we just use the native `hashFiles` function?

Reply:
Actually, scratch that, there are just too many files involved in constructing this cache key, so I think it's cleaner to separate it out and write the key to a file as is currently done.